k8s/upgrade/src/bin/upgrade-job/common/error.rs

```diff
@@ -582,6 +582,36 @@ pub(crate) enum Error {
         args: Vec<String>,
         std_err: String,
     },
+
+    /// Error for when we fail to read the entries of a directory.
+    #[snafu(display("Failed to read the contents of directory {}: {}", path.display(), source))]
+    ReadingDirectoryContents {
+        source: std::io::Error,
+        path: PathBuf,
+    },
+
+    /// Error for when the 'crds' directory inside a helm chart directory is either not a
+    /// directory, or it does not exist.
+    #[snafu(display("Helm chart 'crds' directory {} is invalid", path.display()))]
+    InvalidHelmChartCrdDir { path: PathBuf },
+
+    /// Error for when CRD creation fails.
+    #[snafu(display("Failed to create CustomResourceDefinition '{}': {}", name, source_err))]
+    CreateCrd {
+        source_err: kube::Error,
```

Comment:
usually these are called just `source`

Reply:
I just wanted to use this as a hack because the source field isn't exposed when building it (maybe because it doesn't actually exist in the generated type). 'source' is only captured when using the context api, or so I thought. Turns out there's an IntoError trait which let's you feed the source error to the Snafu type.

Using source, instead of source_err.